opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/indicator-test.ts

```
@@ -82,26 +89,72 @@ describe('Indicator resolver standard behavior', () => {
+        pattern_type: 'stix',
+        x_opencti_main_observable_type: ENTITY_DOMAIN_NAME,
+      },
+    };
+    const indicator = await queryAsAdmin({
+      query: CREATE_QUERY,
+      variables: INDICATOR_TO_CREATE,
+    });
+    expect(indicator.data?.indicatorAdd).toBeDefined();
+    expect(indicator.data?.indicatorAdd.name).toEqual(`New name for ${indicatorForTestName}`);
+    expect(indicator.data?.indicatorAdd.observables.edges.length).toEqual(0);
+    expect(indicator.data?.indicatorAdd.id).toEqual(firstIndicatorInternalId);
   });
   it('should indicator loaded by internal id', async () => {
-    const queryResult = await queryAsAdmin({ query: READ_QUERY, variables: { id: indicatorInternalId } });
+    const queryResult = await queryAsAdmin({ query: READ_QUERY, variables: { id: firstIndicatorInternalId } });
     expect(queryResult).not.toBeNull();
     expect(queryResult.data?.indicator).not.toBeNull();
-    expect(queryResult.data?.indicator.id).toEqual(indicatorInternalId);
+    expect(queryResult.data?.indicator.id).toEqual(firstIndicatorInternalId);
     expect(queryResult.data?.indicator.toStix.length).toBeGreaterThan(5);
   });
   it('should indicator loaded by stix id', async () => {
     const queryResult = await queryAsAdmin({ query: READ_QUERY, variables: { id: indicatorStixId } });
     expect(queryResult).not.toBeNull();
     expect(queryResult.data?.indicator).not.toBeNull();
-    expect(queryResult.data?.indicator.id).toEqual(indicatorInternalId);
+    expect(queryResult.data?.indicator.id).toEqual(firstIndicatorInternalId);
   });
   it('should list indicators', async () => {
     const queryResult = await queryAsAdmin({ query: LIST_QUERY, variables: { first: 10 } });
-    expect(queryResult.data?.indicators.edges.length).toEqual(4);
+    const indicatorList: [] = queryResult.data?.indicators.edges;
+    expect(indicatorList).toBeDefined();
+    if (indicatorList) {
```

I suggest you do not check this ; you already expect it to be defined earlier so the test will fail explicitly.
it makes the code base more straightforward (dumb and simple I might say).

---

deepscan is asking me to check it or it fails, maybe I'm missing something.... it seems that the expect is not enough